contracts/util/LoanLib.sol

```diff
@@ -0,0 +1,339 @@
+                escrowLoanValue = EscrowInterface(loan.escrow).calculateLoanValue();
+                neededInLendingTokens -= int256(escrowLoanValue);
+            }
+        }
+
+        neededInLendingTokens = neededInLendingTokens.percent(requiredRatio);
+    }
+
+    /**
+        @notice It gets the current liquidation info for a given loan.
+        @param loan The loan to get the info.
+        @param loansContract The loans contract instance for the loan.
+        @return liquidationInfo get current liquidation info for the given loan id.
+     */
+    function getLiquidationInfo(
+        TellerCommon.Loan storage loan,
+        LoansInterface loansContract
+    ) public view returns (TellerCommon.LoanLiquidationInfo memory liquidationInfo) {
+        liquidationInfo.collateralInfo = getCollateralInfo(loan, loansContract);
+        liquidationInfo.amountToLiquidate = getTotalOwed(loan);
+
+        // Maximum reward is the calculated value of required collateral minus the principal owed (see LoanLib.getCollateralNeededInTokens).
+        int256 maxReward = liquidationInfo.collateralInfo.neededInLendingTokens -
+            int256(loan.principalOwed);
+        // Available value to payout the liquidator is the value left in collateral + the escrow value. Since the liquidator paid the amount owed, we subtract only the principal amount owed because the collateral ratio already includes the interest.
+        int256 availableRewardValue = int256(
+            liquidationInfo.collateralInfo.valueInLendingTokens
+        ) +
+            int256(liquidationInfo.collateralInfo.escrowLoanValue) -
+            int256(loan.principalOwed);
+        // If there is more than the maximum reward available, only pay the liquidator the max and leave the rest for the borrower to claim.
+        liquidationInfo.rewardInCollateral = availableRewardValue < maxReward
+            ? availableRewardValue
+            : maxReward;
+
+        liquidationInfo.liquidable =
+            loan.status == TellerCommon.LoanStatus.Active &&
+            (loan.loanStartTime.add(loan.loanTerms.duration) <= now ||
+                (loan.loanTerms.collateralRatio > 0 &&
+                    liquidationInfo.collateralInfo.moreCollateralRequired));
+    }
+
+    /**
+        @notice Make a payment towards the interest and principal for a specified loan.
+        @notice Payments are made towards the interest first.
+        @param loan The loan the payment is for.
+        @param toPay The amount of tokens to pay to the loan.
+    */
+    function payOff(TellerCommon.Loan storage loan, uint256 toPay) public {
+        if (toPay >= loan.interestOwed) {
+            toPay = toPay.sub(loan.interestOwed);
+            loan.interestOwed = 0;
+            if (toPay == 0) {
+                return;
+            }
+        } else {
+            loan.interestOwed = loan.interestOwed.sub(toPay);
+            return;
+        }
+        loan.principalOwed = loan.principalOwed.sub(toPay);
```

Comment:
Since the variables used in line 337 are not used in the "if/else" statements, it might be moved as the first line in the function.
So then, we might remove the `return;` in the else (and the "toPay == 0" condition too).

```
        loan.principalOwed = loan.principalOwed.sub(toPay);
        if (toPay >= loan.interestOwed) {
            toPay = toPay.sub(loan.interestOwed);
            loan.interestOwed = 0;
        } else {
            loan.interestOwed = loan.interestOwed.sub(toPay);
        }
```

What would happen if "toPay > principalOwed". Is it possible?

Reply:
Moving the principal to the bottom allows us to pay back the interest first. This is how traditional loans work, but also the required collateral amount is calculated by subtracting values from the collateral ratio. The collateral ratio that is defined on the loan struct is the required amount to take out the loan but once the loan is taken out we can subtract percentage values to know that the min collateral amount is to keep the loan in good standing.

We can calculate the min collateral required by only using the `principalOwed` value because:
1. We pay off the interest first
1. We subtract the interest rate from the collateral ratio

https://github.com/teller-protocol/teller-protocol-v1/blob/dd0704e9736a940154c45113b202d6842ad83310/contracts/util/LoanLib.sol#L261
https://github.com/teller-protocol/teller-protocol-v1/blob/dd0704e9736a940154c45113b202d6842ad83310/contracts/util/LoanLib.sol#L272-L275